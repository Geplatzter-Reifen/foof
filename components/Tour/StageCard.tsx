import React from "react";
import { DATE, dateFormat, getDurationFormatted } from "@/utils/dateUtil";
import { Stage } from "@/model/model";
import { Button, Card, Layout, Text } from "@ui-kitten/components";
import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";
import { deleteStage } from "@/model/database_functions";
import { StyleSheet, View } from "react-native";
import customStyles from "../../constants/styles";
import { foofDarkTheme } from "@/constants/custom-theme";

export default function StageCard({ stage }: { stage: Stage }) {
  const startedAt: Date = new Date(stage.startedAt);
  let finishedAt: Date | undefined = stage.finishedAt
    ? new Date(stage.finishedAt)
    : undefined;

  const date: string = dateFormat(startedAt, DATE);

  let duration: string | undefined = finishedAt
    ? getDurationFormatted(startedAt, finishedAt)
    : getDurationFormatted(startedAt, new Date(Date.now()));

  const distance: string = stage.distance.toFixed(1);

  const avgSpeed: string = stage.avgSpeed.toFixed(1);

  const Header = () => {
    return (
      <View style={styles.header}>
        <Text category="h6" style={styles.title}>
          {stage.title}
        </Text>
        <Button
          status="basic"
          appearance="ghost"
          onPress={() => deleteStage(stage.id)}
        >
          <FontAwesomeIcon icon="trash" />
        </Button>
      </View>
    );
  };

  return (
    <Layout level="2">
      <Card
        style={{
          ...customStyles.basicCard,
          ...customStyles.basicShadow,
          ...styles.card,
        }}
        header={<Header />}
      >
        <View style={styles.stat}>
          <FontAwesomeIcon
            icon="arrows-left-right"
            size={19}
            color={foofDarkTheme["color-primary-500"]}
            style={styles.icon}
          />
          <Text style={styles.statLabel}>{distance} km</Text>
        </View>
        {duration && (
          <View style={styles.stat}>
            <FontAwesomeIcon
              icon="clock"
              size={19}
              color={foofDarkTheme["color-primary-500"]}
              style={styles.icon}
            />
            <Text style={styles.statLabel}>{duration}</Text>
          </View>
        )}
        <View style={styles.stat}>
          <FontAwesomeIcon
            icon="gauge-high"
            size={19}
            color={foofDarkTheme["color-primary-500"]}
            style={styles.icon}
          />
          <Text style={styles.statLabel}>{avgSpeed} km/h</Text>
        </View>
        <Text appearance="hint" style={styles.date}>
          {date}
        </Text>
      </Card>
    </Layout>
  );
}

const styles = StyleSheet.create({
  card: {
    marginBottom: 15,
  },
  header: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "space-between",
  },
  title: {
    marginTop: 10,
    marginLeft: 15,
  },
  stat: {
    flex: 1,
    flexDirection: "row",
    alignItems: "center",
  },
  icon: {
    marginRight: 10,
  },
  statLabel: {
    fontSize: 17,
  },
  date: {
    marginTop: 5,
  },
});
