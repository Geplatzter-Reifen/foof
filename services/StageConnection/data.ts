import { MapPoint } from "@/utils/locationUtil";

export const flensburg: MapPoint = {
  latitude: 54.789356,
  longitude: 9.434199,
};
export const oberstdorf: MapPoint = {
  latitude: 47.408333325,
  longitude: 10.27931564,
};
export const hannover: MapPoint = {
  latitude: 52.3744779,
  longitude: 9.7385532,
};
export const wuerzburg: MapPoint = {
  latitude: 49.7933723,
  longitude: 9.9309779,
};
export const wiesbaden: MapPoint = {
  latitude: 52.520007,
  longitude: 13.404954,
};
