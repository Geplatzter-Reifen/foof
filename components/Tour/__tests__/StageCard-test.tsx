import { render, screen } from "@/test-utils/test-utils";
import {
  createStage,
  setStageAvgSpeed,
  setStageDistance,
  startStage,
} from "@/services/data/stageService";
import { StageCardForTest as StageCard } from "../StageCard";
import { createTour } from "@/services/data/tourService";
import { createManualStage } from "@/services/tracking";
import { waitFor } from "@testing-library/react-native";

describe("StageCard", () => {
  beforeEach(() => {
    jest
      .spyOn(global.Date, "now")
      .mockImplementation(() => new Date("2024-01-01 21:00").getTime());
  });

  test("renders basic stage correctly", async () => {
    const tour = await createTour("Test Tour");
    const stage = await createStage(
      tour.id,
      "Basic Stage",
      new Date("2024-01-01T10:30").getTime(),
      new Date("2024-01-01T17:45").getTime(),
      false,
    );
    await setStageDistance(stage.id, 56.98765);
    await setStageAvgSpeed(stage.id, 7.86036);
    const view = await waitFor(() => render(<StageCard stage={stage} />));
    expect(view).toMatchSnapshot();
  });

  it("renders correctly for active stage", async () => {
    const tour = await createTour("Test Tour");
    const activeStage = await startStage(tour.id);
    const view = await waitFor(() => render(<StageCard stage={activeStage} />));
    expect(view).toMatchSnapshot();
    expect(screen.getByText("00:00 h")).toBeTruthy(); // seit erstellen (Date.now()) und Dauerberechnung (Date.now()) ist noch kaum Zeit vergangen
  });

  it("renders correctly for manually created stage", async () => {
    const tour = await createTour("Test Tour");
    const manualStage = await createManualStage(
      "Manual Stage",
      "50.3, 8.9",
      "50.8, 8.1",
      new Date("2024-12-03 8:13"),
      new Date("2024-12-03 16:49"),
      tour.id,
    );
    const view = await waitFor(() => render(<StageCard stage={manualStage} />));
    expect(view).toMatchSnapshot();
    expect(screen.getByText("79.3 km")).toBeTruthy();
    expect(screen.getByText("08:36 h")).toBeTruthy();
    expect(screen.getByText("9.2 km/h")).toBeTruthy();
  });
});
