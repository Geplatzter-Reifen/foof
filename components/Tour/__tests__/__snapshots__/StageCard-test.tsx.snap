// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StageCard renders basic stage correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#7e7e7e",
      "borderRadius": 8,
      "borderWidth": 0,
      "elevation": 3,
      "flex": 1,
      "marginBottom": 15,
      "marginHorizontal": 10,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 2,
    }
  }
  testID="stage-card"
>
  <View
    style={
      {
        "backgroundColor": undefined,
        "height": 0,
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      appearance="default"
      category="h5"
      status="basic"
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 22,
            "fontWeight": "800",
          },
          {
            "marginLeft": 15,
            "marginTop": 8,
          },
        ]
      }
    >
      Basic Stage
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "minHeight": 40,
            "minWidth": 40,
            "opacity": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 12,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#494949",
                  "fontSize": 18,
                },
                {
                  "marginHorizontal": 10,
                  "width": "auto",
                },
                {
                  "fontFamily": "FontAwesome6Free-Solid",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "900",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "minHeight": 40,
            "minWidth": 40,
            "opacity": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 12,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#494949",
                  "fontSize": 18,
                },
                {
                  "marginHorizontal": 10,
                  "width": "auto",
                },
                {
                  "fontFamily": "FontAwesome6Free-Solid",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "900",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    appearance="default"
    style={
      [
        {
          "backgroundColor": "#b8b8b8",
          "height": 1,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      [
        {
          "flexShrink": 1,
        },
        {
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        },
      ]
    }
  >
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 22,
                },
                {
                  "color": "#FF6E14",
                  "width": "auto",
                },
                {
                  "fontFamily": "FontAwesome6Free-Solid",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "900",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              {
                "fontSize": 18,
                "marginLeft": 8,
              },
            ]
          }
        >
          0.0 km
        </Text>
      </View>
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 22,
                },
                {
                  "color": "#FF6E14",
                  "width": "auto",
                },
                {
                  "fontFamily": "FontAwesome6Free-Solid",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "900",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              {
                "fontSize": 18,
                "marginLeft": 8,
              },
            ]
          }
        >
          7h 15m
        </Text>
      </View>
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 22,
                },
                {
                  "color": "#FF6E14",
                  "width": "auto",
                },
                {
                  "fontFamily": "FontAwesome6Free-Solid",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "900",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              {
                "fontSize": 18,
                "marginLeft": 8,
              },
            ]
          }
        >
          0.0 km/h
        </Text>
      </View>
    </View>
    <Text
      appearance="hint"
      category="p1"
      status="basic"
      style={
        [
          {
            "color": "#494949",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
          },
          {
            "marginTop": 9,
          },
        ]
      }
    >
      01.01.2024 10:30
    </Text>
  </View>
</View>
`;

exports[`StageCard renders correctly for active stage 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#7e7e7e",
      "borderRadius": 8,
      "borderWidth": 0,
      "elevation": 3,
      "flex": 1,
      "marginBottom": 15,
      "marginHorizontal": 10,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 2,
    }
  }
  testID="stage-card"
>
  <View
    style={
      {
        "backgroundColor": "#FF6E14",
        "height": 4,
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      appearance="default"
      category="h5"
      status="basic"
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 22,
            "fontWeight": "800",
          },
          {
            "marginLeft": 15,
            "marginTop": 8,
          },
        ]
      }
    >
      Etappe 1
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "minHeight": 40,
            "minWidth": 40,
            "opacity": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 12,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#494949",
                  "fontSize": 18,
                },
                {
                  "marginHorizontal": 10,
                  "width": "auto",
                },
                {
                  "fontFamily": "FontAwesome6Free-Solid",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "900",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "minHeight": 40,
            "minWidth": 40,
            "opacity": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 12,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#494949",
                  "fontSize": 18,
                },
                {
                  "marginHorizontal": 10,
                  "width": "auto",
                },
                {
                  "fontFamily": "FontAwesome6Free-Solid",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "900",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    appearance="default"
    style={
      [
        {
          "backgroundColor": "#b8b8b8",
          "height": 1,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      [
        {
          "flexShrink": 1,
        },
        {
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        },
      ]
    }
  >
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 22,
                },
                {
                  "color": "#FF6E14",
                  "width": "auto",
                },
                {
                  "fontFamily": "FontAwesome6Free-Solid",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "900",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              {
                "fontSize": 18,
                "marginLeft": 8,
              },
            ]
          }
        >
          0.0 km
        </Text>
      </View>
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 22,
                },
                {
                  "color": "#FF6E14",
                  "width": "auto",
                },
                {
                  "fontFamily": "FontAwesome6Free-Solid",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "900",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              {
                "fontSize": 18,
                "marginLeft": 8,
              },
            ]
          }
        >
          0h 0m
        </Text>
      </View>
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 22,
                },
                {
                  "color": "#FF6E14",
                  "width": "auto",
                },
                {
                  "fontFamily": "FontAwesome6Free-Solid",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "900",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              {
                "fontSize": 18,
                "marginLeft": 8,
              },
            ]
          }
        >
          NaN km/h
        </Text>
      </View>
    </View>
    <Text
      appearance="hint"
      category="p1"
      status="basic"
      style={
        [
          {
            "color": "#494949",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
          },
          {
            "marginTop": 9,
          },
        ]
      }
    >
      01.01.2024 21:00
    </Text>
  </View>
</View>
`;

exports[`StageCard renders correctly for manually created stage 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#7e7e7e",
      "borderRadius": 8,
      "borderWidth": 0,
      "elevation": 3,
      "flex": 1,
      "marginBottom": 15,
      "marginHorizontal": 10,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 2,
    }
  }
  testID="stage-card"
>
  <View
    style={
      {
        "backgroundColor": undefined,
        "height": 0,
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      appearance="default"
      category="h5"
      status="basic"
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 22,
            "fontWeight": "800",
          },
          {
            "marginLeft": 15,
            "marginTop": 8,
          },
        ]
      }
    >
      Manual Stage
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "minHeight": 40,
            "minWidth": 40,
            "opacity": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 12,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#494949",
                  "fontSize": 18,
                },
                {
                  "marginHorizontal": 10,
                  "width": "auto",
                },
                {
                  "fontFamily": "FontAwesome6Free-Solid",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "900",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "minHeight": 40,
            "minWidth": 40,
            "opacity": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 12,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#494949",
                  "fontSize": 18,
                },
                {
                  "marginHorizontal": 10,
                  "width": "auto",
                },
                {
                  "fontFamily": "FontAwesome6Free-Solid",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "900",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    appearance="default"
    style={
      [
        {
          "backgroundColor": "#b8b8b8",
          "height": 1,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      [
        {
          "flexShrink": 1,
        },
        {
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        },
      ]
    }
  >
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 22,
                },
                {
                  "color": "#FF6E14",
                  "width": "auto",
                },
                {
                  "fontFamily": "FontAwesome6Free-Solid",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "900",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              {
                "fontSize": 18,
                "marginLeft": 8,
              },
            ]
          }
        >
          79.3 km
        </Text>
      </View>
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 22,
                },
                {
                  "color": "#FF6E14",
                  "width": "auto",
                },
                {
                  "fontFamily": "FontAwesome6Free-Solid",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "900",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              {
                "fontSize": 18,
                "marginLeft": 8,
              },
            ]
          }
        >
          8h 36m
        </Text>
      </View>
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 22,
                },
                {
                  "color": "#FF6E14",
                  "width": "auto",
                },
                {
                  "fontFamily": "FontAwesome6Free-Solid",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "900",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              {
                "fontSize": 18,
                "marginLeft": 8,
              },
            ]
          }
        >
          9.2 km/h
        </Text>
      </View>
    </View>
    <Text
      appearance="hint"
      category="p1"
      status="basic"
      style={
        [
          {
            "color": "#494949",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
          },
          {
            "marginTop": 9,
          },
        ]
      }
    >
      03.12.2024 08:13
    </Text>
  </View>
</View>
`;
